repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/CaseTypeRepository.java

```diff
@@ -18,8 +18,10 @@
     @Query(SELECT_LATEST_CASE_TYPE_ENTITY_FOR_REFERENCE)
     Optional<CaseTypeEntity> findCurrentVersionForReference(@Param("caseTypeReference") String caseTypeReference);
 
-    @Query("select c from CaseTypeEntity c where c.version in (select max(cm.version) from CaseTypeEntity cm "
-        + "where cm.reference=c.reference) and c.jurisdiction.reference=:jurisdictionReference")
+    @Query(value = "SELECT * FROM case_type c LEFT JOIN case_type cm "
+        + "ON c.reference = cm.reference AND c.version < cm.version "
+        + "JOIN jurisdiction j ON c.jurisdiction_id =j.id "
+        + "WHERE j.reference =:jurisdictionReference AND cm.version IS NULL",  nativeQuery = true)
```

Comment:
@kaanaktas  - I'm not sure if this query is fetching indeed the same resultset. Let's have an interactive chat, maybe I'm missing something.

Reply:
Nice trick actually, thank you.